client/src/components/HomePage.jsx

```diff
@@ -1,63 +1,93 @@
 import React from 'react';
 import Introduction from './Introduction.jsx';
 import Picture from './Picture.jsx';
+import TourPage from './TourPage.jsx';
+import ListView from './ListView.jsx';
 import $ from 'jquery';
 import queryString from 'query-string';
 
-
 class HomePage extends React.Component {
   constructor(props) {
     super(props);
     this.state = {
       image: {},
+      tourView: false,
+      listView: false,
     };
+
     this.getImage = this.getImage.bind(this);
+    this.handlePageViewClick = this.handlePageViewClick.bind(this);
+    this.handleListViewClick = this.handleListViewClick.bind(this);
   }
 
   componentDidMount() {
     let parsed = queryString.parse(location.search);
     let currentId = Number(parsed.id);
     if (currentId) {
-      this.getImage(Number(parsed.id));      
+      this.getImage(Number(parsed.id));
     } else {
       this.getImage(1);
     }
   }
 
   getImage(id) {
-    $.get(`/images/${id}`, function(data) {
+    $.get(`/images/${id}`, function (data) {
     })
       .done((data) => {
-        this.setState({image: data});
-        console.log('CLIENT GET IMAGE OBJ :: ', data.image);        
+        this.setState({ image: data });
+        console.log('CLIENT GET IMAGE OBJ :: ', data.image);
       })
       .fail((error) => {
         return;
       });
   }
 
   saveImageToList(id) {
-    $.patch(`/images/${id}`, function(data) {
+    $.patch(`/images/${id}`, function (data) {
     })
       .done((data) => {
-        let arr = [];
-        arr.push(data);
-        this.setState({image: data});
-        console.log('CLIENT PATCH IMAGE OBJ :: ', data.image);        
+        this.setState({ image: data });
+        console.log('CLIENT PATCH IMAGE OBJ :: ', data.image);
       })
       .fail((error) => {
         console.log('CLIENT PATCH FAILED');
         return;
       });
   }
 
+  handlePageViewClick() {
+    this.setState((prevState) => {
+      return { tourView: !prevState.tourView };
+    });
+  }
+
+
+  handleListViewClick() {
+    this.setState((prevState) => {
+      return { listView: !prevState.listView };
+    });
+  }
+
   render() {
     return (
-      <div className="home-page">
-        <Introduction image={this.state.image}/>  
-        { this.state.image.image && 
-         <Picture image={this.state.image}/>           
-        }         
+      <div className='home-page'>
+        <Introduction image={this.state.image} />
+        {this.state.image.image &&
```

Comment:
consider rethinking the naming convention for whatever is stored at `this.state.image.image` the repetition of `image` is confusing to me.

Reply:
agreed, was looking at that too. will refactor